lib/ui/platform_dispatcher.dart

```diff
@@ -1642,31 +1642,53 @@ class FrameTiming {
+/// instance, when the OS calls
+/// [`Activity.onPause`](https://developer.android.com/reference/android/app/Activity#onPause()),
+/// Flutter will enter the `inactive` state, but when Android calls
+/// [`Activity.onStop`](https://developer.android.com/reference/android/app/Activity#onStop()),
+/// Flutter enters the `paused` state. See the individual state documenation for
+/// descriptions of what they mean on each platform.
 ///
 /// See also:
 ///
-///  * [WidgetsBindingObserver], for a mechanism to observe the lifecycle state
-///    from the widgets layer.
+/// * [WidgetsBindingObserver], for a mechanism to observe the lifecycle state
+///   from the widgets layer.
 enum AppLifecycleState {
-  /// The application is visible and responding to user input.
+  /// The application is visible and responsive to user input.
+  ///
+  /// On Android, this state corresponds to an app or the Flutter host view
+  /// having focus while in Android's "resumed" state. It is possible for the
+  /// Flutter app to be in the [inactive] state while still being in Android's
+  /// "resumed" state if the app has lost focus, but hasn't had
+  /// [`Activity.onPause`](https://developer.android.com/reference/android/app/Activity#onPause())
+  /// called on it.
   resumed,
 
   /// The application is in an inactive state and is not receiving user input.
   ///
   /// On iOS, this state corresponds to an app or the Flutter host view running
-  /// in the foreground inactive state. Apps transition to this state when in
-  /// a phone call, responding to a TouchID request, when entering the app
+  /// in the foreground inactive state. Apps transition to this state when in a
+  /// phone call, responding to a TouchID request, when entering the app
   /// switcher or the control center, or when the UIViewController hosting the
   /// Flutter app is transitioning.
   ///
-  /// On Android, this corresponds to an app or the Flutter host view running
-  /// in the foreground inactive state.  Apps transition to this state when
-  /// another activity is focused, such as a split-screen app, a phone call,
-  /// a picture-in-picture app, a system dialog, or another view.
+  /// On Android, this corresponds to an app or the Flutter host view running in
+  /// Android's paused state (i.e.
+  /// [`Activity.onPause`](https://developer.android.com/reference/android/app/Activity#onPause())
+  /// has been called), or in Android's "resumed" state (i.e.
+  /// [`Activity.onResume`](https://developer.android.com/reference/android/app/Activity#onResume())
+  /// has been called) but it has lost window focus. Examples of when apps
+  /// transition to this state include when the app is partially obscured or
+  /// another activity is focused, such as: a split-screen app, a phone call, a
+  /// picture-in-picture app, a system dialog, another view, when the
+  /// notification window shade is down, or the application switcher is visible.
```

Comment:
are we sure about this for all OEMs? The Android docs aren't very clear about it

Reply:
I haven't tested all OEMs, I'm just assuming that they function similarly to PIxel when it comes to the window shade and app switcher. Keeping focus on the app in either of those two cases would be pretty confusing, I would think, but I suppose it could be done. What subset of OEMs would we need to test?

Reply:
Make sure to also test multi-window: https://developer.android.com/guide/topics/large-screens/multi-window-support#lifecycle

Reply:
I don't know how to make a multi-window app where the Activity has multiple windows in it where one is a Flutter window. I'll look for an example.

I did test it with the split screen functionality (two activities up at the same time), and that works correctly. Clicking on one of them focuses the input, and sets the Flutter state to resumed, and clicking away sets it to inactive: it used to ignore that and not send anything.

Reply:
I believe that in Android multi-window mode (which includes split-screen mode), each of the windows has to be a different Activity.

Typically this means the windows come from unrelated apps (and [the doc](https://developer.android.com/guide/topics/large-screens/multi-window-support) repeatedly refers to "two apps", "another app", etc.), because most apps only ever show one activity at a time. But I just tried a quick demo with one of the few apps I use that cheerfully makes multiple activities at a time, namely Google Sheets: open the app, open one sheet, go back to the list of sheets, open a second sheet, put the two sheets in split-screen mode. And split-screen mode worked great for those two activities from the same app.

So after testing split-screen mode where one of the activities is a Flutter window, the next more complex edge case would be split-screen mode where both of the activities are Flutter windows from the same app. (I'm not sure if that's currently possible, though, or if it's still in the future — I know you've been working on Flutter multi-window support.)

Reply:
Okay, I tried the Realme (I couldn't find an Oppo), and although it's app switcher takes it all the way to `paused` (Android's `onStop` state), unlike the Pixel one (which just notifies of focus loss, and leaves the app in the "onResume" Android state), it does correct notification. I suspect it does this because it doesn't provide a live view of the app in the app switcher, just a snapshot. The changes also work fine in split screen mode, and with the window shade.